static/js/containers/CollectionDetailPage.js

```diff
@@ -112,10 +123,22 @@ class CollectionDetailPage extends React.Component {
         </div>
         <div className="tools">
           {
-            collection.is_admin &&
-            <a onClick={ showDialog.bind(this, DIALOGS.NEW_COLLECTION) }>
-              <i className="material-icons">settings</i>
-            </a>
+            collection.is_admin && [
+              <DropboxChooser
+                key="upload"
+                appKey={SETTINGS.dropbox_key}
```

Comment:
Only users with admin permission are allowed to upload a file from dropbox, so we could limit this key to members of the admin group. 

However, I don't think there's much risk associated with leaking this key. It's our app key right? That's just how dropbox identifies us in order to manage our access to their APIs. I think the worst case of leaking it is that someone else uses it and DOS's us by going over quota. It doesn't access any of our resources.

Reply:
The issue is that a user can be an admin for a specific set of collections, but that app key is obviously constant for all collections. We _could_ just tack on the dropbox key to any request for a specific collection where the user happens to be an admin, but that's a hack and could introduce some complications in the front end logic. Given that this is app requires a login and there's low risk even if the app key is leaked, I'm fine just leaving it as-is